pkg/goldpinger/client.go

```diff
@@ -54,6 +55,64 @@ func CheckNeighboursNeighbours(ctx context.Context) *models.CheckAllResults {
 	return CheckAllPods(ctx, SelectPods())
 }
 
+// CheckCluster does a CheckNeighboursNeighbours and analyses results to produce a binary OK or not OK
+func CheckCluster(ctx context.Context) *models.ClusterHealthResults {
+
+	start := time.Now()
+	output := models.ClusterHealthResults{
+		GeneratedAt: strfmt.DateTime(start),
+		OK:          true,
+	}
+	selectedPods := SelectPods()
+
+	// precompute the expected set of nodes
+	expectedNodes := []string{}
+	for _, peer := range selectedPods {
+		expectedNodes = append(expectedNodes, peer.HostIP)
+	}
+	sort.Strings(expectedNodes)
+
+	// get the response we serve for check_all
+	checkAll := CheckAllPods(ctx, selectedPods)
+
+	// we should at the very least have a response from ourselves
+	if len(checkAll.Responses) < 1 {
+		output.OK = false
+	}
+	for _, resp := range checkAll.Responses {
+		// 1. check that all nodes report OK
+		if *resp.OK {
+			output.NodesHealthy = append(output.NodesHealthy, resp.HostIP.String())
+		} else {
+			output.NodesUnhealthy = append(output.NodesUnhealthy, resp.HostIP.String())
+			output.OK = false
+		}
+		output.NodesTotal++
+		// 2. check that all nodes report the expected peers
+		// on error, there might be no response from the node
+		if resp.Response == nil {
+			output.OK = false
```

Comment:
You will still need a `continue` or put the next section in an `else`. Otherwise you will get a panic when you hit `resp.Response.PodResults`